Bump mentoring and its dependency xblock-utils for Harvard

bradenmacdonald opened the pull request:
**Description**: This updates the requirements for edx.org to use the new "problem builder" (mentoring v2) block, which is required for upcoming Harvard courses. 

The two updated repositories references were reviewed as [OSPR-419](https://openedx.atlassian.net/browse/OSPR-419) (https://github.com/open-craft/xblock-mentoring/pull/4) and [OSPR-422](https://openedx.atlassian.net/browse/OSPR-422) (https://github.com/edx/xblock-utils/pull/5).

**Partner information**: hosted on edx.org (Harvard)

**Sandbox**: http://sandbox.opencraft.com:18010/

**Merge deadline**: TBD

edx-webhook commented:
Thanks for the pull request, @bradenmacdonald! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=7534

bradenmacdonald commented:
@antoviaque FYI. Please review/edit as needed.

bradenmacdonald commented:
@antoviaque One option to consider is moving mentoring v2 to a separate repo called "problem-builder", without making any other changes (so no further review is needed), which would allow mentoring v1 and v2 to be installed side-by-side, and upgraded on a per-course basis.

antoviaque commented:
test this please

antoviaque commented:
@bradenmacdonald Thanks for this! :+1: There are test failures, but they seem unrelated, I've restarted a build.

And good idea for the renaming - that would allow to keep the previous version for the archived courses, without risking to introduce bugs that we wouldn't see there. That would work for me. @sarina what do you think of it? (And also, do you want an OSPR ticket for this?)

sarina commented:
@antoviaque @bradenmacdonald nah - just merge this once you get a passing build. I think master may be broken at the moment - so perhaps try rebasing later today.

bradenmacdonald commented:
OK. This is now blocked by https://github.com/open-craft/problem-builder/pull/1 . I'd like to merge that first, and then we can change this PR so it installs the new XBlock alongside the old one, rather than replacing it. Then the warning in this PR description will no longer apply.

antoviaque commented:
@sarina Cf @bradenmacdonald 's remarks about the migration - I think renaming now makes sense to avoid having to care about migrating the previous Harvard courses when we push this to edx.org. It would be less risky and less work, but that would require reviewing/merging this: https://github.com/open-craft/problem-builder/pull/1 - it's a lengthy patch but it just renames the XBlock where "mentoring" was used. Do you agree that it would be better to do that now, rather than migrating the old courses? And if so how do you want to handle the review?

edx-webhook commented:
Thanks for the pull request, @bradenmacdonald! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=7534

sarina commented:
@antoviaque that makes sense.

I'll take a quick look over the problem-builder PR and give it thumbs.

bradenmacdonald commented:
Rebased and updated hash so the new block is installed in parallel with the old version. Tests are now passing. From my perspective this should be good to go.

sarina commented:
@bradenmacdonald - yeah, we've given thumbs for all aspects of this work, yes? If this is ready, feel free to merge. Be sure it's what you want to go to prod - the release candidate will be cut Monday morning at 10 EDT.

bradenmacdonald commented:
OK, thanks @sarina. Merging now so it will be included in the release.
